### PR TITLE
feat(cli): restore --initial-files-dir flag for agents create/update

### PR DESCRIPTION
**Pull request by chaliy**
## What
Re-adds the `--initial-files-dir` flag to `agents create` and `agents update` CLI commands, allowing bulk upload of local directory contents as read-only `initial_files` on agent definitions.

## Why
The flag was removed in #1028 as collateral of dropping `serde_yaml` from the CLI. The underlying server API still supports `initial_files`, but there was no ergonomic CLI way to populate them from a directory.

## How
- Added `--initial-files-dir <dir>` to both `Create` and `Update` subcommands (requires `--file`)
- `glob_initial_files()` uses the `ignore` crate to recursively walk the directory, respecting `.gitignore` and skipping hidden/binary files
- `parse_agent_file_as_json()` parses Markdown/YAML/JSON agent files into a `serde_json::Value`, injects the collected files as `initial_files`, then sends as JSON to the server import API
- Files get `/workspace/` prefix paths, text encoding, `is_readonly: true`
- Re-added `serde_yaml` to CLI deps (already in workspace)

## Risk
- Low
- Only affects the `--initial-files-dir` code path; existing `--file` without the flag is unchanged (still sends `text/plain`)

## Security
- Threat categories reviewed: TM-FS, TM-API, TM-DOS
- TM-FS: `canonicalize()` resolves symlinks, `strip_prefix` ensures relative paths, hidden files skipped. CLI runs with user privileges so no privilege escalation.
- TM-API: No new endpoints. Server import API already accepts JSON. Server-side `validate_import_file_size` provides size limits.
- TM-DOS: `ignore` crate respects `.gitignore` to scope walks. Server validates payload size.
- No auth/authz/tenant/SQL/LLM changes.

## Checklist
- [x] Tests added or updated (7 new unit tests: parsing JSON/YAML/Markdown, directory globbing, edge cases)
- [x] Backward compatibility considered (additive flag, no breaking changes)
- [x] Security review performed against relevant threat model categories
- [x] All review comments addressed (code change or written reasoning)

### CODE REVIEW

File: crates/cli/src/commands/agents.rs [hunk 1]
```diff
@@ -232,6 +279,114 @@ async fn import_from_file(
...
+fn glob_initial_files(dir: &str) -> Result<Vec<CollectedFile>> {
+    let base =
+        std::fs::canonicalize(dir).with_context(|| format!("Cannot resolve directory: {}", dir))?;
+    if !base.is_dir() {
+        anyhow::bail!("Not a directory: {}", base.display());
+    }
+
+    let walker = ignore::WalkBuilder::new(&base)
+        .hidden(true) // respect hidden files filter
+        .git_ignore(true) // respect .gitignore
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/cli/src/commands/agents.rs]: Fixed in 3d15d09. Added git_global(false) and git_exclude(false) for predictable behavior.

File: crates/cli/src/commands/agents.rs [hunk 2]
```diff
@@ -232,6 +279,114 @@ async fn import_from_file(
...
+        .build();
+
+    let mut files = Vec::new();
+    for entry in walker {
+        let entry = entry.context("Failed to read directory entry")?;
+        let path = entry.path();
+        if !path.is_file() {
+            continue;
+        }
+
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/cli/src/commands/agents.rs:314]: Fixed in 3d15d09. Each file path is now canonicalized and verified to still start with the base directory, preventing symlink escapes. Added test.

File: crates/cli/src/commands/agents.rs [hunk 3]
```diff
@@ -232,6 +279,114 @@ async fn import_from_file(
...
+        // Skip hidden files/directories (dot-prefixed relative components).
+        // The ignore crate handles this too, but belt-and-suspenders.
+        if rel
+            .components()
+            .any(|c| c.as_os_str().to_string_lossy().starts_with('.'))
+        {
+            continue;
+        }
+        let workspace_path = format!("/workspace/{}", rel.display());
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/cli/src/commands/agents.rs]: Fixed in 3d15d09. Path separators are now normalized to POSIX-style for workspace paths.

File: crates/cli/src/commands/agents.rs [hunk 4]
```diff
@@ -232,6 +279,114 @@ async fn import_from_file(
...
+
+        // If there's a markdown body and no system_prompt in frontmatter, use it
+        if !body.is_empty()
+            && (obj.get("system_prompt").is_none()
+                || obj["system_prompt"].as_str().is_none_or(|s| s.is_empty()))
+        {
+            obj["system_prompt"] = serde_json::Value::String(body.to_string());
+        }
+
+        return Ok(obj);
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/cli/src/commands/agents.rs]: Fixed in 3d15d09. Frontmatter parsing now uses line-based iteration, requiring the delimiter on its own line.

File: crates/cli/src/commands/agents.rs [hunk 5]
```diff
@@ -232,6 +279,114 @@ async fn import_from_file(
...
+        let mut obj: serde_json::Value =
+            serde_yaml::from_str(frontmatter).context("Failed to parse YAML frontmatter")?;
+
+        // If there's a markdown body and no system_prompt in frontmatter, use it
+        if !body.is_empty()
+            && (obj.get("system_prompt").is_none()
+                || obj["system_prompt"].as_str().is_none_or(|s| s.is_empty()))
+        {
+            obj["system_prompt"] = serde_json::Value::String(body.to_string());
+        }
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/cli/src/commands/agents.rs]: Fixed in 3d15d09. Both frontmatter and YAML paths now validate the parsed value is a JSON object before indexing. Added test for non-object rejection.
